src/types/schema.ts

```diff
@@ -1,65 +1,105 @@
 import { UUID } from 'crypto';
 
-export interface CaseListing {
-  id: UUID;
-  title: string;
-  summary: string;
-  country: string;
-  client_location: string;
-  legal_server_id: number;
-  hours_per_month: number;
-  num_months: number;
-  languages: string[];
-  is_remote: boolean;
-  in_court: boolean;
-  needs_attorney: boolean;
-  needs_interpreter: boolean;
-  upcoming_date: string;
-  relief_sought: string;
-  active: boolean;
+// table field types
+export type ImmigrationLawExperienceEnum = 'LOW' | 'MEDIUM' | 'HIGH';
+
+export type AdjudicatingAgencyEnum =
+  | '9TH_CIRCUIT'
+  | 'BIA'
+  | 'IMMIGRATION_COURT'
+  | 'ICE'
+  | 'USCIS';
+
+export type RoleEnum =
+  | 'ATTORNEY'
+  | 'INTERPRETER'
+  | 'LEGAL_FELLOW'
+  | 'TRANSLATOR';
+
+export type ListingTypeEnum =
+  | 'CASE'
+  | 'LIMITED_ASSISTANCE'
+  | 'TRANSLATION_REQUEST';
+
+// join table rows
+export interface ProfileLanguage {
+  user_id: UUID;
+  iso_code: string;
+  can_read: boolean;
+  can_write: boolean;
 }
 
-export interface LimitedAssistance {
-  id: UUID;
-  summary: string;
-  languages: string[];
-  country: string;
-  experience_level: string;
-  deadline: string;
-  interest_ids: UUID[];
+export interface ProfileRole {
+  user_id: UUID;
+  role: RoleEnum;
 }
 
-export interface TranslationRequest {
+export interface CaseLanguage {
+  listing_id: UUID;
+  iso_code: string;
+}
+
+export interface CaseRelief {
+  listing_id: UUID;
+  relief_code: string;
+}
+
+// sql table rows
+export interface CaseListing {
   id: UUID;
-  summary: string;
-  languages: string[];
-  interest_ids: UUID[];
+  title?: string;
+  summary?: string;
+  country?: string;
+  client_location?: string;
+  legal_server_id: number;
+  hours_per_month?: number;
```

Comment:
do we want `hours_per_month` to be optional? I think David added a tag on the case listing card component that corresponds to a case's anticipated hrs/month. Are we fine with not displaying that tag for all cards if some cases don't have that info, or should we make it required to show the hrs/month data for all case listing cards?

Reply:
great point!! the hours per month is an additional field that is going to be added to LegalServer, and so there is the possibility of it being `null` if someone forgets to input it on LegalServer. generally we're making all LegalServer fields nullable because there's always the possibility of someone forgetting to fill them out on there.

@davidqing6432 wondering if you have ideas for modifications to the case listing card in the case that some of these fields are null?